Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/manager/impl/ConceptManager.java

```diff
@@ -628,4 +634,15 @@ public ConceptEntry getOriginalConceptEntry(String id) {
     public List<ConceptEntry> getAllConcepts() {
         return client.getAllConcepts();
     }
+
+	@Override
+	public List<ConceptEntry> getAllConceptsByStatus(ReviewStatus status) {
+		List<ConceptEntry> result = new ArrayList<>();
+		
+		for(ReviewRequest request: requestsManager.getReviewByStatus(status)) {
+			result.add(getConceptEntry(request.getConceptId()));
+		}
```

Comment:
have you tried using a join query, something like
`SELECT c FROM ConceptEntry c, ReviewRequest r WHERE c.id = r.concept_id AND r.status = :status`?

I'm really not an expert on this, but I"m sure there is a way to let the database do all of this and avoid multiple database calls, especially since you'll want to paginate.

Reply:
You are right. But the trick is that not all the concepts are in our MySQL database. We have concepts coming from both the MySQL and the wordnet DB.

Reply:
I think we need to change the way reviews can be created then. Basically, before a someone can add a review request to a concept that concept has to be turned into a "real" concept (conceptwrapper).

Reply:
?